resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -250,11 +250,21 @@ private[spark] object Config extends Logging {
       .stringConf
       .createOptional
 
+  private val podConfValidator =
+    "^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$".r.pattern
+
   val KUBERNETES_EXECUTOR_POD_NAME_PREFIX =
     ConfigBuilder("spark.kubernetes.executor.podNamePrefix")
-      .doc("Prefix to use in front of the executor pod names.")
+      .doc("Prefix to use in front of the executor pod names. Note that pod names must consist" +
+        " of lower case alphanumeric characters, '-' or '.', and must start and end with an" +
+        " alphanumeric character (e.g. 'example.com', regex used for validation is:" +
+        s" ${podConfValidator.toString}")
```

Comment:
Hi, @yaooqinn . Thank you, but this is still incomplete because we don't check `contain at most 63 characters` rule.

If they change, this will become wrong again. So, we had better give the official reference because this is not defined by Apache Spark. We just follow K8s's rule. Technically, the official definition is `RFC 1123` in K8s document, https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names, isn't it?

Reply:
> Hi, @yaooqinn . Thank you, but this is still incomplete because we don't check contain at most 63 characters rule.
yes we shall do the length checking too

> If they change, this will become wrong again. So, we had better give the official reference because this is not defined by Apache Spark. We just follow K8s's rule. Technically, the official definition is RFC 1123 in K8s document, https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names, isn't it?

Make sense. and at least  they seemed already change the error message :) https://github.com/kubernetes/kubernetes/pull/94182


FYI, just found a `go` implementation for k8s API validation -  https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go, I think we can follow it in this PR and later for other resource checks.

Reply:
+1 for following their implementation.

Reply:
As `ExecutorPodsAllocator` in k8s uses `AtomicLong` to track executor Id, we may have the longest name might be `$podNamePrefix-exec-9223372036854775807`. the max length can only be 38.